manager-parent/manager-core/src/main/resources/db/migration/0.2/V0.2.1__create_v2_db.sql

```diff
@@ -0,0 +1,75 @@
+CREATE TABLE BRIDGE_V2 (
+    id                 varchar(255) NOT NULL PRIMARY KEY,
+
+    customer_id        varchar(255) NOT NULL,
+    organisation_id    varchar(255) NOT NULL,
+    owner              varchar(255) NOT NULL,
+    subscription_id    varchar(255) NOT NULL,
+
+    name               varchar(255) NOT NULL,
+    cloud_provider     varchar(255) NOT NULL,
+    region             varchar(255) NOT NULL,
+
+    endpoint           varchar(255),
+    shard_id           varchar(255) NOT NULL,
+
+    submitted_at       timestamp    NOT NULL,
+    published_at       timestamp,
+
+    error_id           integer,
+    error_uuid         varchar(255),
+
+    version            integer      NOT NULL default 0,
+    generation         integer      NOT NULL default 0,
+
+    unique (customer_id, name)
+);
+
+
+CREATE TABLE PROCESSOR_V2 (
+    id                 varchar(255) NOT NULL PRIMARY KEY,
+
+    name               varchar(255) NOT NULL,
+    flows              jsonb        NOT NULL,
+
+    owner              varchar(255) NOT NULL,
+
+    submitted_at       timestamp    NOT NULL,
+    published_at       timestamp,
+
+    version            integer      NOT NULL default 0,
+    generation         integer      NOT NULL default 0,
+
+    error_id           integer,
+    error_uuid         varchar(255),
+
+    bridge_id          varchar(255) NOT NULL,
+
+    unique (bridge_id, name),
+    constraint fk_bridge foreign key (bridge_id) references BRIDGE_V2 (id)
+);
+
+CREATE TABLE OPERATION (
```

Comment:
I don't understand the goal of this `OPERATION` table 🤔 
As far as I see this is tied to `CONDITION` but a condition can be created directly by the shard operator so when this operation is created/linked?

Reply:
See 
![Screenshot from 2022-11-24 11-01-11](https://user-images.githubusercontent.com/18282531/203832252-5ac480ac-3fd2-43fb-8bb6-e98d5e3c0a8c.png)

for the DB schema

Reply:
The `Operation` aims to keep track of the user's request (like `create`, `update`, `delete`) and the set of conditions is linked to this `Operation`

Reply:
Does that mean that every condition returned by the operator is then saved in the DB spitting the status section in the different conditions "linked" by the `Operation`?
What if the operator returns a "failed" condition that is not triggered by any user input (i.e. a system error or some external conditions that have changed and now the resource is not working anymore)?

Reply:
Thinking at what you wrote I have also another questions, do we plan to cleanup the `Operation` over time?
I.e. a processor is:
- created
- updated
- updated
- deleted

Do we plan to keep the whole "history"? I expect we only need the current/latest

Reply:
The idea is that we store a new `Condition` for every dependency deployed by the control plane (as well as if it fails), and we store all the conditions returned by the data plane. The top-level status of the resource is then calculated at runtime when the user interacts with the REST API (for example, if all the conditions are `Ready`, then the resource is `Ready`. If there is at least a `Failed` condition, then the resource is in the `Failed` state. @manstis prepared a document to describe the logic that we are going to implement for this). 

Currently we keep a 1:1 mapping between the managed resource and the operation - meaning we keep just the latest request of the user. But we thought this data model could be easy to be extended to keep the entire history of the user's requests and their results

Reply:
See https://docs.google.com/document/d/1TM_ev8RV_Vft9YusIVDRe0zm7aHE2G-VHszFxYOpBRs/edit

Reply:
We were not keen on the notion of keeping an _operation_ i.e. something done to a resource; on the resource itself.

An `Operation` is applied to a `Resource` by `Work`. The outcome of the `Operation` is a status (`Condition`'s).